### DISCUSSION TITLE
event metadata / placeholders

### DISCUSSION THREAD

**Discussion by listerr** [pretix/pretix#3009] [Ideas] [answered]
Currently there is **"Item metadata"**
This allows custom fields to be added to items (products).

* Accessed via Settings -> General -> Item metadata

> "You can here define a set of metadata properties (i.e. variables) that you can later set for your items and re-use in places like ticket layouts."

* Within Products, under each product/item configuration, the general metadata appear here with values as defined in  Settings -> General -> Item metadata, but looks like it can be overridden in each product if needed.

These can then be used in the ticket pdf layout.

In reality, I use these as "event global" metadata (things specific to the event can be printed on the ticket layout, without having to hack it manually as text into every ticket layout.), but they are not available elsewhere, for example in email placeholders. 

### Feature Suggestion:

- Add **Event** metadata: global variables that can be used everywhere:
   - In the ticket layout
   - Certificate layout
   - Email out voucher placeholders
   - Send out email placeholders

- Some method to access as placeholders e.g.` { var.<name> }`, so:  `{ var.my_custom_field }` , `{ var.some_other_field }` to avoid possible long cluttered list as the help text in the send email forms.

**Comment by raphaelm** [comment, marked as answer]
This exists. You can define the parameters in the settings on organizer level :)

**Comment by listerr** [comment]
Hmmm. I'll have to check this out.

Is it available as placeholders in emails?

I need variables to be settable for each event, not the same thing for every event.

When I create a new event, I usually clone it from a previous event (we only do two or three a year, rather than a regular series of events.)

What I tend to find is not everyone reads or realises there's important information specific to each event embedded in the ticket PDF: It needs to go in the email so they read it before they travel. The first time they open the PDF will be on their phone when they go to show the ticket on arrival.

For us it's usually specific to each venue, for example what to do when you arrive, which room to go to, if the venue requires photo ID, public transport and parking directions etc.


Thanks!



Rob

*From: *Raphael Michel ***@***.***>
*To: *pretix/pretix ***@***.***>
*CC: *Rob Lister ***@***.***>; Author ***@***.***>
*Date: *5 Jan 2023 19:09:41
*Subject: *Re: [pretix/pretix] event metadata / placeholders (Discussion #3009)

**Reply by raphaelm** [reply]
yup that works

**Reply by listerr** [reply]
Wow! I never knew this existed!

I would never have thought to look in the organizer account settings. The user guide doesn't really mention it.

I guess because if there are no event meta fields set up in organizer level, nothing appears in event _Settings -> Basics_ to hint that this is even a possibility.  **Meta data** only shows up when something is configured.

e.g. in "Customer and attendee data", you see **Custom fields:** Manage questions, so you know that custom fields/questions are a thing, and also where to configure those. :)

Fields show up as email placeholders `{ meta_<name> }`, so this is very useful.

Thanks :)
